packages/mini-oxygen/src/vite/worker-entry.ts

```diff
@@ -91,27 +92,22 @@ let runtime: ViteRuntime;
  */
 function fetchEntryModule(publicUrl: URL, env: ViteEnv) {
   if (!runtime) {
-    let onHmrRecieve: ((payload: HMRPayload) => void) | undefined;
+    let onHmrRecieve: ((payload: HotPayload) => void) | undefined;
 
     let hmrReady = false;
     connectHmrWsClient(publicUrl, env)
       .then((hmrWs) => {
         hmrReady = !!hmrWs;
         hmrWs?.addEventListener('message', (message) => {
           if (!message.data) return;
-          const data: HMRPayload = JSON.parse(message.data.toString());
+          const data: HotPayload = JSON.parse(message.data.toString());
 
           if (!data) return;
-
           if (data.type === 'update') {
             // Invalidate cache synchronously without revalidating the
             // module to avoid hanging promises in workerd
             for (const update of data.updates) {
-              runtime.moduleCache.invalidateDepTree([
-                // Module IDs are absolute from root
-                update.path.replace(/^\.\//, '/'),
-                ...(update.ssrInvalidates ?? []),
-              ]);
+              runtime.import(update.acceptedPath);
```

Comment:
This could lead to tricky bugs in HMR. I would encourage testing HMR in different situations and see if something fails (both server and browser code).
Not saying this change is not correct, but this area was tricky to get right so it would be better to double check (if you haven't yet).